site-packages/integralstor/manifest_status.py

```diff
@@ -160,53 +225,61 @@ def get_cpu_cores():
+            mem_total = {}
+            mem_free = {}
             for line in lines:
-                match = re.match(
-                    'MemTotal\s*:\s+([\S]+)\s+(\S+)$', line.strip())
-                if match:
-                    grps = match.groups()
-                    if grps:
-                        mem_total = {}
-                        mem_total['unit'] = grps[1]
-                        mem_total['value'] = int(grps[0])
-                        if mem_total['unit'] == 'kB':
-                            mem_total['unit'] = 'GB'
-                            mem_total['value'] = float(
-                                mem_total['value']) / 1024 / 1024
-                        elif mem_total['unit'] == 'mB':
-                            mem_total['unit'] = 'GB'
-                            mem_total['value'] = float(
-                                mem_total['value']) / 1024
-                match = re.match(
-                    'MemFree\s*:\s+([\S]+)\s+(\S+)$', line.strip())
-                if match:
-                    grps = match.groups()
-                    if grps:
-                        mem_free = {}
-                        mem_free['unit'] = grps[1]
-                        mem_free['value'] = int(grps[0])
-                        if mem_free['unit'] == 'kB':
-                            mem_free['unit'] = 'GB'
-                            mem_free['value'] = float(
-                                mem_free['value']) / 1024 / 1024
-                        elif mem_free['unit'] == 'mB':
-                            mem_free['unit'] = 'GB'
-                            mem_free['value'] = float(
-                                mem_total['value']) / 1024
-                if mem_total and mem_free:
-                    break
-        if not mem_total and not mem_free:
-            raise Exception('Error retrieving memory information')
-        ret['mem_total'] = mem_total
-        ret['mem_free'] = mem_free
+                if type in ['info', 'all']:
+                    match = re.match(
+                        'MemTotal\s*:\s+([\S]+)\s+(\S+)$', line.strip())
+                    if match:
+                        grps = match.groups()
+                        if grps:
+                            mem_total['unit'] = grps[1]
+                            mem_total['value'] = int(grps[0])
+                            if mem_total['unit'] == 'kB':
+                                mem_total['unit'] = 'GB'
+                                mem_total['value'] = int(float(
+                                    mem_total['value']) / 1024 / 1024)
+                            elif mem_total['unit'] == 'mB':
+                                mem_total['unit'] = 'GB'
+                                mem_total['value'] = int(float(
+                                    mem_total['value']) / 1024)
+                    if mem_total:
+                        ret['mem_total'] = mem_total
+                        break
```

Comment:
This will break the control out off the loop even when `type` == 'all', causing it to raise an error because `mem_free` will not have any value.

Reply:
Found this in later testing.. Actually, there are two places where the breaks should not be there..